ctapipe/io/magiceventsource.py

```diff
@@ -0,0 +1,1337 @@
+
+
+import glob
+import re
+
+import uproot
+import numpy as np
+import scipy.interpolate
+
+from astropy import units as u
+from astropy.coordinates import Angle
+from astropy.time import Time
+from ctapipe.io.eventsource import EventSource
+from ctapipe.io.containers import DataContainer, TelescopePointingContainer, WeatherContainer
+from ctapipe.instrument import TelescopeDescription, SubarrayDescription, OpticsDescription, CameraGeometry
+import gzip
+import struct
+
+__all__ = ['MAGICEventSourceHDF5', 'MAGICEventSourceROOT']
+
+
+class MAGICEventSourceHDF5(EventSource):
+    """
+    EventSource for MAGIC raw data converted to hdf5.
+
+    This class utilises `h5py` to read the hdf5 file, and stores the
+    information into the event containers.
+    """
+    _count = 0
+
+    def __init__(self, config=None, tool=None, **kwargs):
+        super().__init__(config=config, tool=tool, **kwargs)
+
+        try:
+            import h5py
+        except ImportError:
+            msg = "The `h5py` python module is required to access the MAGIC data or MCs"
+            self.log.error(msg)
+            raise
+
+        self.h5py = h5py
+        self.file = h5py.File(self.input_url)
+
+#         if MAGICEventSource._count > 0:
+#             self.log.warn("Only one MAGIC raw event_source allowed at a time. "
+#                           "Previous hdf5 file will be closed.")
+#             self.pyhessio.close_file()
+#         MAGICEventSource._count += 1
+
+
+    @staticmethod
+    def is_compatible(file_path):
+        import h5py
+        # check general format:
+        if not h5py.is_hdf5(file_path):
+            return False
+        # crude check if hdf5 file contains MAGIC raw data:
+        with h5py.File(file_path, "r") as file:
+            intrument_attr = file.attrs['instrument']
+            if intrument_attr.tostring() == b"MAGIC":
```

Comment:
what is `f.attrs['instrument']` supposed to be? I tried writing both a `str` and a `bytes`, neither have a `tostring` method and why compare with bytes `v'MAGIC'`?